src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -42,25 +55,56 @@ public JSONArray getSupportedLocalesFromRemoteVIP() {
         return dao.getSupportedLocalesFromRemoteVIP();
     }
 
+    /**
+     * Retrieves translated messages of all components of a product in the requested locale (See the dto object).
+     *
+     * @return translated messages of all components of a product locale specified in the dto object
+     */
     public List<Map> getAllComponentTranslation() {
         List<Map> list = new ArrayList<Map>();
-        Object[] locales = {};
-        Object[] components = {};
-        if (VIPCfg.getInstance().getMessageOrigin() == DataSourceEnum.VIP) {
-            locales = this.getSupportedLocalesFromRemoteVIP().toArray();
-            components = this.getComponentsFromRemoteVIP()
-                    .toArray();
-        }
-        for (Object locale : locales) {
-            for (Object component : components) {
-                dto.setComponent(((String) component).trim());
-                dto.setLocale(LocaleUtility.fmtToMappedLocale((String) locale).toString().trim());
-                Map<String, String> retMap = new ComponentService(dto).getMessages().getCachedData();
-                if (retMap != null) {
-                    list.add(retMap);
+        LocaleDTO localeDTO = new LocaleDTO(dto.getProductID(), dto.getVersion());
+        Map<String, String> locales = new LocaleService(localeDTO).getSupportedLanguages();
+        List<String> components = this.getComponents(VIPCfg.getInstance().getMsgOriginsQueue().iterator());
+        if (locales != null) {
+            for (String languageTag : locales.keySet()) {
+                for (Object component : components) {
+                    MessagesDTO msgDTO = new MessagesDTO(((String) component).trim(), LocaleUtility.fmtToMappedLocale(Locale.forLanguageTag(languageTag)).toString().trim(),
+                            dto.getProductID(), dto.getVersion());
+                    Iterator<Locale> fallbackLocalesIter = LocaleUtility.getFallbackLocales().iterator();
+                    Map<String, String> retMap = new ComponentService(msgDTO).getMessages(fallbackLocalesIter).getCachedData();
```

Comment:
It's unnecessary to do local fallback here because the locales used here are supported no doubt.